chia/farmer/farmer_api.py

```diff
@@ -565,30 +567,28 @@ async def request_signed_values(self, full_node_request: farmer_protocol.Request
             full_node_request.quality_string
         ]
 
-        foliage_block_data: Optional[SignatureRequestSourceData] = None
-        foliage_transaction_block_data: Optional[SignatureRequestSourceData] = None
-        include_source_data = False
+        message_data: Optional[List[Optional[SignatureRequestSourceData]]] = None
 
         if full_node_request.foliage_block_data is not None:
-            include_source_data = True
-            foliage_block_data = SignatureRequestSourceData(
-                uint8(SigningDataKind.FOLIAGE_BLOCK_DATA), bytes(full_node_request.foliage_block_data)
-            )
-
-        if full_node_request.foliage_transaction_block_data is not None:
-            assert foliage_block_data
-            include_source_data = True
-            foliage_transaction_block_data = SignatureRequestSourceData(
-                uint8(SigningDataKind.FOLIAGE_TRANSACTION_BLOCK),
-                bytes(full_node_request.foliage_transaction_block_data),
-            )
+            message_data = [
+                SignatureRequestSourceData(
+                    uint8(SigningDataKind.FOLIAGE_BLOCK_DATA), bytes(full_node_request.foliage_block_data)
+                ),
+                None
+                if full_node_request.foliage_transaction_block_data is None
+                else SignatureRequestSourceData(
+                    uint8(SigningDataKind.FOLIAGE_TRANSACTION_BLOCK),
+                    bytes(full_node_request.foliage_transaction_block_data),
+                ),
+            ]
 
         request = harvester_protocol.RequestSignatures(
             plot_identifier,
             challenge_hash,
             sp_hash,
             [full_node_request.foliage_block_data_hash, full_node_request.foliage_transaction_block_hash],
-            message_data=[foliage_block_data, foliage_transaction_block_data] if include_source_data else None,
+            message_data=message_data,
```

Comment:
the `message_data` field is suppose to be identical to the previous logic, right?

Reply:
Yes, should be the same logic, just less lines of code